dotCMS/src/main/java/com/dotcms/cms/login/LoginServiceAPIImpl.java

```diff
@@ -0,0 +1,569 @@
+            /* Custom Code */
+            User user = null;
+            if (Validator.isEmailAddress(decryptedId))
+                user = userAPI.loadByUserByEmail(decryptedId, systemUser, false);
+            else
+                user = userAPI.loadUserById(decryptedId, systemUser, false);
+            /* End of Custom Code */
+            try {
+                String userName = (authByEmail) ? user.getEmailAddress() : user.getUserId();
+
+                return _doLogin(userName, null, true, request, response, true);
+            } catch (Exception e) { // $codepro.audit.disable logExceptions
+                SecurityLogger.logInfo(this.getClass(),
+                                "An invalid attempt to login (No user found) from IP: "
+                                                + request.getRemoteAddr() + " :  " + e);
+
+                return false;
+            }
+        } catch (Exception e) {
+            SecurityLogger.logInfo(this.getClass(), "Auto login failed (No user found) from IP: "
+                            + request.getRemoteAddr() + " :  " + e);
+
+            if (useCASLoginFilter) {
+                String decryptedId = PublicEncryptionFactory.decryptString(encryptedId);
+                Logger.info(this.getClass(),
+                                "Try to retrieve user from LDAP/CAS with id: " + decryptedId);
+                User newUser = CASAuthUtils.loadUserFromLDAP(decryptedId);
+
+                if (UtilMethods.isSet(newUser)) {
+                    User user = null;
+
+                    try {
+                        user = (authByEmail)
+                                        ? userAPI.loadByUserByEmail(decryptedId, systemUser, false)
+                                        : userAPI.loadUserById(decryptedId, systemUser, false);
+
+
+                        String userIdFromCAS = (String) request.getSession(false)
+                                        .getAttribute("edu.yale.its.tp.cas.client.filter.user");
+
+                        if (UtilMethods.isSet(userIdFromCAS)) {
+                            CASAuthUtils.setUserValuesOnSession(user, request, response, true);
+                        }
+
+                        return true;
+
+                    } catch (Exception ex) {
+                        return false;
+                    }
+                } else
+                    Logger.info(this.getClass(), "Unable to retrieve user from LDAP/CAS with id: "
+                                    + decryptedId);
+            }
+
+            doLogout(request, response);
+
+            return false;
+
+        }
+    }
+
+
+
+    @Override
+    public boolean doCookieLogin(final String encryptedId, final HttpServletRequest request,
+                    final HttpServletResponse response) {
+        // note: keep in mind we are doing BE and FE login, not sure if this is right
+        final boolean doCookieLogin = _doCookieLogin(encryptedId, request, response);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "doCookieLogin". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)